### DISCUSSION TITLE
Failed to compile. Type error: Module '"@prisma/client"' has no exported member 'UserStatus'.

### DISCUSSION THREAD

**Discussion by David-orovs** [vercel/community#3494] [Help]
### Summary

I am trying to deploy on vercel but i keep getting this "

Failed to compile.
--
19:07:05.476 |  
19:07:05.477 | ./src/modules/controllers/user.ts:9:10
19:07:05.477 | Type error: Module '"@prisma/client"' has no exported member 'UserStatus'.
19:07:05.477 |  
19:07:05.477 | 7 \| import { NextApiRequest, NextApiResponse } from 'next';
19:07:05.477 | 8 \| import { NextHandler } from 'next-connect';
19:07:05.477 | >  9 \| import { UserStatus, Currency } from '@prisma/client';
19:07:05.477 | \|          ^
19:07:05.477 | 10 \| import { joiValidate } from '../helpers/utilities';
19:07:05.477 | 11 \| import {
19:07:05.477 | 12 \|   addUserProfileSchema,
19:07:05.512 | error Command failed with exit code 1.
19:07:05.512 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
19:07:05.535 | Error: Command "yarn run build" exited with 1"
And i don't get this issue when i run npm run build locally, it works normally but fails when i deploy to vercel

### Steps to Reproduce

just deployment bug of running npm run build locally and it works, but fails on deployment

**Comment by MDRietveld** [comment]
I have a similar issue. The import @prisma/client, but a typeerror "Does not exist"


Type error: Property 'PrismaClientKnownRequestError' does not exist on type 'typeof Prisma'.
--
15:34:41.698 |  
15:34:41.699 | 46 \|       } catch (e) {
15:34:41.699 | 47 \|         if (
15:34:41.699 | > 48 \|           e instanceof Prisma.PrismaClientKnownRequestError &&
15:34:41.699 | \|                               ^
15:34:41.699 | 49 \|           e.code === "P2002"
15:34:41.699 | 50 \|         ) {
15:34:41.699 | 51 \|           return {
15:34:41.738 | error Command failed with exit code 1.

Manual CLI builds work, automatic preview deploy doesn't work

**Comment by MDRietveld** [comment]
@David-orovs I think I found the problem. Could you try the following:
Go to Vercel --> Settings --> General.
At Build & Development Settings, I expect you to have Framework Preset "Next.js"

Overwrite your build command with this:
`prisma generate && prisma migrate deploy && next build`

This wil generate the required imports and apply migrations

(this fixes my issues)
